### DISCUSSION TITLE
Forking Renovate not running on new repositories

### DISCUSSION THREAD

**Discussion by Neenu1995** [renovatebot/renovate#10289] [Request Help] [answered]
We have [Forking Renovate](https://github.com/apps/forking-renovate) installed org-wide, however, all the recently created repos seem to be ignored by the bot. These repositories were created as private repositories and then made public. New repositories that were created as public seem to be found.

Examples:

https://github.com/googleapis/java-dms
https://github.com/googleapis/java-service-usage

Is there a way to trigger a rescan across our org or trigger a repository to be watched?

cc: @justinbeckwith @danoscarmike

**Comment by rarkins** [comment]
I reset the status for all installed repositories yesterday which have status=disabled. They should have all run a few times already by now

**Reply by danoscarmike** [reply]
Thanks @rarkins. Is there something we're doing that disables these? Anything we should be doing to avoid this in future (we have a bunch of new repos coming)?

**Reply by rarkins** [reply]
I think we need to locate the first run where Renovate detects the repo, parses it, and declares it as disabled. Disabled repos aren't tried any more than weekly. It's probably due to lack of dependencies at the time. I'm not sure if any easy solution to it but will think about it.

**Reply by viceice** [reply]
Couldn't renovate simply react to push events to schedule a run, even if it was disabled?

**Reply by rarkins** [reply]
That would cause thousands of unnecessary runs per day if we reacted to every push event from disabled repos. Maybe OK for renovate.json though

**Reply by danoscarmike** [reply]
IIUC within a week these kinds of repos will be re-parsed and if deps exist then would be reenabled. Is that right? If yes, that's probably good enough.

**Reply by Neenu1995** [reply]
Hi,
I made a couple repos public 2 days ago.

https://github.com/googleapis/java-apigee-connect
https://github.com/googleapis/java-life-sciences

Renovate has not run on them yet. These have some outdated dependencies in them, so it is not because of a lack of dependencies to update. Is there a way to enable these repos automatically? Maybe when the repo is made public or when triggered with a specific push to renovate.json?

**Reply by rarkins** [reply]
@Neenu1995 sure, I can try to work out a way where your normal workflow would be "caught" by Renovate so we don't need to do it manually.

What's the typical ordering? Is it the following?
1. Repo is created a private
2. Forking Renovate is installed
3. Dependencies are added
4. Repo is made public
5. `renovate.json` is added manually?

**Reply by rarkins** [reply]
Looking at the logs, Renovate ran 3 times before now in `googleapis/java-apigee-connect` with the key result being:
1. "Repository is empty - skipping","time":"2021-06-04T17:30:57.378Z"
2. "Cannot fork repository - skipping","time":"2021-06-04T17:31:26.861Z" [It got a 404 when attempting to Fork, maybe it was still private?]
3. "Cannot fork repository - skipping","time":"2021-06-06T19:54:25.525Z" [same as above]

I'm guessing that (1) was triggered at time of install (webhook) and that (2) was probably another commit or webhook that occurred a minute or so after. (3) was when we ran a batch job on all disabled repos on the weekend.

Can you confirm when it was made public so we can correlate that with the above timestamps?

**Reply by Neenu1995** [reply]
The ordering is as follows:

1. Repo is created private
2. Library is pushed to the repo with dependencies and renovate.json
3. Few days later, repo is made public

Forking renovate is installed org wide.

From the log above for `java-apigee-connect`, 

- (1) is when the private repo was created
-  (2) is when The library was pushed to the repo with the dependencies and renovate.json
- the repo was made public on June 8th at 9.17 am PST

**Reply by Neenu1995** [reply]
Friendly ping!

**Reply by rarkins** [reply]
Thanks! I read this first on mobile and it got marked as read.

It seems that the event that Renovate needs to react to is the "repo is made public" event, because that's the first time it can properly run.

https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#public

**Reply by Neenu1995** [reply]
Yes. That should work. Can the  "repo is made public" event be used to trigger Renovate bot?

**Reply by rarkins** [reply, marked as answer]
Yes, it can and should now be working. I've added that "public" event handling then tested with a private repo I switched to public and the PRs began right away.

**Reply by Neenu1995** [reply]
Awesome! Thank you.

**Reply by Neenu1995** [reply]
Hi,
Couple of libraries were made public before the fix came in. Can you manually trigger those please?
https://github.com/googleapis/java-eventarc
https://github.com/googleapis/java-gke-connect-gateway

**Reply by rarkins** [reply]
Done, they should be run again soon

**Reply by Neenu1995** [reply]
Thank you :)
